cli/flox/src/commands/pull.rs

```diff
@@ -406,23 +407,23 @@ impl Pull {
             },
 
             // Failed to _build_ the environment due to an incompatible package
-            EnvironmentError::Core(ref core_err @ CoreEnvironmentError::BuildEnv(_))
-                if core_err.is_incompatible_package_error() =>
-            {
+            EnvironmentError::Core(
+                ref core_err @ CoreEnvironmentError::BuildEnv(BuildEnvError::Realise2 { .. }),
+            ) => {
                 debug!(
                     "environment contains package incompatible with the current system: {err}",
```

Comment:
With the catalog this code path is pretty unlikely, right? Maybe there's a flake package that fails to build on this system, or there's a conflict between two packages on one system but not another? But the package itself wouldn't necessarily be incompatible, right?
```suggestion
                    "failed to build environment: {err}",
```